### PR TITLE
position sensitive roi-pool support (R-FCN)

### PR DESCRIPTION
**Pull request by terrychenism**
I reimplemented a psroi-pooling based on R-FCN(https://github.com/daijifeng001/R-FCN). Combined by rcnn example, you can easily implement R-FCN by mxnet！

### CODE REVIEW

File: src/common/gpu_util.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#include <curand.h>
+
+
+
+template <typename Dtype>
+inline __device__ Dtype mxnet_gpu_atomic_add(const Dtype val, Dtype* address);
+
+template <>
+inline __device__
+float mxnet_gpu_atomic_add(const float val, float* address) {
```
Comment (piiswrong) [inline src/common/gpu_util.h]: is this wraper necessary?

Comment (terrychenism) [inline src/common/gpu_util.h]: Yes, maybe we can merge into  "cuda_utils.h"

File: src/common/gpu_util.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * Copyright (c) 2016 by Contributors
+ * \file cuda_utils.h
+ * \brief CUDA debugging utilities.
+ */
+
+#ifndef MXNET_UTIL_GPU_UTIL_H_
+#define MXNET_UTIL_GPU_UTIL_H_
+
+
+#if MXNET_USE_CUDA
```
Comment (piiswrong) [inline src/common/gpu_util.h]: use defined(__CUDACC__) instead

**Comment by piiswrong** [comment]
@precedenceguo @antinucleon @winstywang Could you review this?

@terrychenism Any updates?

**Comment by sbodenstein** [comment]
@piiswrong: we would like to make use of this layer, but it seems like the author has dissapeared. What is the procedure we should use: create a new pull request with a new implementation?

**Comment by piiswrong** [comment]
@sbodenstein If you have made substantial modifications I think it's ok to create a new PR. Please @ the author in your new pr

**Comment by piiswrong** [comment]
@precedenceguo Any comment on this?

**Comment by terrychenism** [comment]
@piiswrong  plz don't merge. I will open a new pr for mx-rfcn and psroi layer.

**Comment by mli** [comment]
close this pr now since a new pr will be made.

**Pull request closed without merging**